### DISCUSSION TITLE
Milvus search with ids

### DISCUSSION THREAD

**Discussion by ufuk23** [milvus-io/milvus#4666] [Q&A and General discussion]
Is it possible to search with given ids as well as vectors using rest api?

For example:

POST
http://localhost:19121/collections/art/vectors
{
  "search": {
      "topk": 3,
      "vectors": [[0,0,0,0,0,0,1,0,0,0,0,0,0,0,0.01,0,0,0,0,0,0,0.2,0,0,0,0,0,0,0,0.3,0,0,0,0,0,0,0,0,0,0,0]],
      "params": {"nprobe": 16},
      **"ids":["5", "23", "123", "1234", "334"]**
  }
}

**Comment by jingkl** [comment]
Can you explain why there is a need for this? What does it look like in an actual business scenario?

**Reply by ufuk23** [reply]
For example, we need a scenario that works on Solr search engine and get the filtered results with IDs from Solr to search their feature vectors in Milvus by the same IDs. We have the filtered IDs coming from Solr and need to search their feature vectors. We don't want to search the whole collection of the milvus db, we need a subset of the collection by specified IDs. How could we do that?

**Reply by jingkl** [reply]
We currently have an interface Get_entity_by_id(collection_name, ids, timeout=None)，If you don't want to use this interface you can go to GitHub and raise an issue

**Reply by ufuk23** [reply]
Yes but it is not the same, it doesn't work while searching, I'll raise an issue.
